nixos/doc/manual/development/activation-script.section.md

```diff
@@ -0,0 +1,67 @@
+# Activation script {#sec-activation-script}
+
+The activation script is a bash script called to activate the new
+configuration which resides in a NixOS system in `$out/activate`. Since its
+contents are dependant on your system configuration, the contents may differ.
+This chapter explains how the script works in general, and some common NixOS
+snippets. Please be aware that the script is executed on every boot and system
+switch, so tasks that can be performed in other places should be performed
+there (for example creating a directory for service in the `preStart` of the
+service).
```

Comment:
```suggestion
there (for example creating directories for services in the service definition itself, either via systemd's `StateDirectory`, or `preStart`).
```

Let's make it more clear where this should happen, and encourage StateDirectory over PreStart.